extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -782,20 +816,33 @@ export class AmpLightboxViewer extends AMP.BaseElement {
 
     this.cleanupEventListeners_();
 
-    // Reset the state of the description box
-    this.descriptionBox_.classList.remove('hide');
+    this.vsync_.mutate(() => {
+      // Reset the state of the description box
+      this.descriptionBox_.classList.remove('hide');
 
-    // If there's gallery, set gallery to display none
-    this.container_.removeAttribute('gallery-view');
+      // If there's gallery, set gallery to display none
+      this.container_.removeAttribute('gallery-view');
+
```

Comment:
Right now, we remove and recreate the gallery on demand. What we could do is do is the same as what is being done with carousels, which is to make one for each lightbox group and show / hide the correct one.

Reply:
would be good as code comment right above the this.gallery_ = null; line

Reply:
This has already been fixed actually, so this comment is no longer relevant. We already create once and show hide subsequently for gallery.